### PR TITLE
Feature/users

### PR DESCRIPTION
**Pull request by yremmet**

### CODE REVIEW

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java [hunk 1]
```diff
@@ -27,6 +31,13 @@
 
     public static final String VERSION = "version";
 
+    private HashMap<String, Object> craeteUser(String username, String password){
```
Comment (marberger) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:28]: Typo: craeteUser -> createUser

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java [hunk 2]
```diff
@@ -45,52 +56,55 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
...
-            mongodb_exporter.put("user", usernamePasswordCredential.getUsername());
-            mongodb_exporter.put("password", usernamePasswordCredential.getPassword());
-            auth.put("user", usernamePasswordCredential.getUsername());
-            auth.put("password", usernamePasswordCredential.getPassword());
-
-            if (!replset.containsKey("keyfile")) {
-                replset.put("keyfile", new RandomString(1024).nextString());
-            }
 
-            if (!properties.containsKey(REPLICA_SET_NAME)) {
```
Comment (marberger) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:79]: Removed but not added again? On purpose?

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java [hunk 3]
```diff
@@ -45,52 +56,55 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
...
+                exporter_properties.put("database", "admin");
+
+                backup_properties.put("user", backupCredential.getUsername());
+                backup_properties.put("password", backupCredential.getPassword());
             }
 
-            replset.put("name", properties.get(REPLICA_SET_NAME));
-            serviceInstance.getParameters().put("replicaSet", properties.get(REPLICA_SET_NAME));
 
-            if (properties.containsKey(DATA_DIR)) {
```
Comment (marberger) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:86]: Removed but not added again? On purpose?

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java [hunk 4]
```diff
@@ -45,52 +56,55 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
...
-                mongodb.put(DATA_DIR, properties.get(DATA_DIR));
-            }
-
-            if (properties.containsKey(VERSION)) {
```
Comment (marberger) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:90]: Removed but not added again? On purpose?

File: osb-service/src/main/resources/bosh/manifest.yml [hunk 5]
```diff
@@ -2,56 +2,66 @@
...
+          admin_users:
+          - username: ((root_credentials.username))
+            password: ((root_credentials.password))
+          - username: ((exporter_credentials.username))
+            password: ((exporter_credentials.password))
+          backup_users:
+          - username: ((backup_credentials.username))
+            password: ((backup_credentials.password))
+          replica-set:
+            keyfile: ((replicaSetKey))
```
Comment (marberger) [inline osb-service/src/main/resources/bosh/manifest.yml:49]: Where is the new keyfile coming from?

**Review by marberger — COMMENTED** [review]
A typo and a few questions. But generally looks good.

File: osb-service/src/main/resources/bosh/manifest.yml (hunk 5 above)
Comment (yremmet) [reply osb-service/src/main/resources/bosh/manifest.yml:49]: From Credhub or if the Broker is configured without Credhub he will replace it.

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java (hunk 4 above)
Comment (yremmet) [reply osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:90]: Not on purpose will add it back again.

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java (hunk 3 above)
Comment (yremmet) [reply osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:86]: Not on purpose but there is no value in choosing a directory.

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java (hunk 2 above)
Comment (yremmet) [reply osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java:79]: Not on purpose will add it back again.

**Comment by marberger** [comment]
LGTM, too. Merging ...

**Pull request merged**